File: Algorithm/Code_Materials/python_source/charpter11_bitmap/level2/multiply.py
# coding=utf-8

def multiply(A, B):
    min_num = min(A, B)
    max_num = max(A, B)
    ans = 0
    for i in range(min_num.bit_length()):
        if min_num & 1:
            ans += max_num
        min_num >>= 1
        max_num += max_num
    return ans


if __name__ == '__main__':
    print(multiply(7, 3))  # 输出: 3

File: Algorithm/Leetcode/Code_Materials/c_or_cpp/charpter11_bitmap/level2/multiply.cpp
#include <stdio.h>
#include <stdlib.h>

int multiply(int A, int B) {
    int min = A < B ? A : B;
    int max = A > B ? A : B;
    int ans = 0;
    for (int i = 0; min != 0; i++) {
        if ((min & 1) == 1) {
            ans += max;
        }
        min >>= 1;
        max += max;
    }
    return ans;
}


int main() {
   
 
    int test=multiply(12,13);
    
    return 0;
}